tests/unit/pages/Editor.spec.js

```diff
@@ -103,4 +121,867 @@ describe("Editor.vue", () => {
+    // hence harcoding here
+    const MINIMUM_QUESTION_TIMESTAMP = 0.6;
+
+    // set some valid items and select the first item
+    await wrapper.setData({
+      items: dummyItemsCopy.data,
+      currentItemIndex: 0,
+    });
+
+    // the correct timestamp value should be populated in itemTimestamps
+    expect(wrapper.vm.itemTimestamps[0]).toBe(dummyItemsCopy.data[0].time);
+
+    // update items with an invalid time value -> will call itemTimestamps watcher
+    // the invalid time value should be fixed back to `MINIMUM_QUESTION_TIMESTAMP`
+    let updatedDummyItems = dummyItemsCopy.data;
+    updatedDummyItems[0].time = 0.1;
+    await wrapper.setData({ items: updatedDummyItems });
+
+    expect(wrapper.vm.items[0].time).toBe(MINIMUM_QUESTION_TIMESTAMP);
+  });
+
+  it("handles title updation correctly", async () => {
+    const checkAndSavePlio = jest.spyOn(Editor.methods, "checkAndSavePlio");
+    const wrapper = mount(Editor);
+
+    await wrapper.setData({ plioTitle: "title for plio" });
+    expect(wrapper.vm.loadedPlioDetails.plioTitle).not.toBe(
+      wrapper.vm.plioTitle
+    );
+    expect(checkAndSavePlio).toHaveBeenCalled();
+  });
+
+  it("computes the itemImage property correctly", async () => {
+    const wrapper = mount(Editor);
+
+    await wrapper.setData({
+      items: dummyItemsCopy.data,
+      currentItemIndex: 0,
+    });
+
+    expect(wrapper.vm.itemImage).toBe(null);
+
+    const imageURL = "test url";
+    const dummyItemsWithImage = dummyItemsCopy.data;
+    dummyItemsWithImage[0].details.image = {
+      id: 56,
+      url: imageURL,
+      alt_text: "Image",
+      created_at: "2021-07-02T12:58:41.683683Z",
+      updated_at: "2021-07-02T12:58:41.684174Z",
+    };
+
+    await wrapper.setData({
+      items: dummyItemsWithImage,
+    });
+
+    expect(wrapper.vm.itemImage).toBe(imageURL);
+  });
+
+  it("computes itemType correctly", async () => {
+    const wrapper = mount(Editor);
+    await wrapper.setData({
+      currentItemIndex: 0,
+      items: dummyItemsCopy.data,
+    });
+    expect(wrapper.vm.itemType).toBe(null);
+    await wrapper.setData({
+      isItemSelected: true,
+    });
+    expect(wrapper.vm.itemType).toBe(dummyItemsCopy.data[0].type);
+  });
+
+  it("computes correctOptionInex correctly", async () => {
+    const wrapper = mount(Editor);
+    await wrapper.setData({
+      items: dummyItemsCopy.data,
+      currentItemIndex: 0,
+    });
+    expect(wrapper.vm.correctOptionIndex).toBe(
+      dummyItemsCopy.data[0].details.correct_answer
+    );
+  });
+
+  it("renders publish button tooltip correctly", async () => {
+    const wrapper = mount(Editor);
+    await wrapper.setData({
+      status: "published",
+      hasUnpublishedChanges: true,
+    });
+    expect(wrapper.vm.publishButtonTooltip).toBe(
+      "Click to publish your changes"
+    );
+    expect(
+      wrapper.find('[data-test="publishButton"]').element.$_ptooltipValue
+    ).toBe("Click to publish your changes");
```

Comment:
can we load this from the language file and read it from there?

Reply:
Not sure
In all of our testing code we have done this way only